compiler/il/OMRResolvedMethodSymbol.cpp

```diff
@@ -1233,9 +1233,12 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
                && comp->supportsInduceOSR()
                && !self()->cannotAttemptOSRDuring(siteIndex, comp);
 
-            optimizer = TR::Optimizer::createOptimizer(comp, self(), true);
-            previousOptimizer = comp->getOptimizer();
-            comp->setOptimizer(optimizer);
+            if (!(comp->getOption(TR_SkipIlgenOpts)))
+               {
+               optimizer = TR::Optimizer::createOptimizer(comp, self(), true);
+               previousOptimizer = comp->getOptimizer();
+               comp->setOptimizer(optimizer);
+               }
```

Comment:
I was surprised to discover there isn't a means to disable this already.  I did find two unused functions `allowIlGenOptimizations()` that nothing in OMR nor any known downstream project uses, but I'd have to dig into the intent of those functions to know if, ultimately, those are the right functions to use.

In the meantime, for your case, can you rename your option `TR_DisableIlgenOpts` to be more consistent with other options for disabling features?